src/controllers/cartController.ts

```diff
@@ -99,4 +99,37 @@ export default class cartController {
       .status(201)
       .json({ message: 'Product added to cart successfully', cart })
   }
+
+  
+  /**
+ * Clears the active cart for the user
+ * @param {Request} req - Express request object
+ * @param {Response} res - Express response object
+ * @returns {Promise<Response>} Promise that resolves to an Express response
+ */
+  static async clearCart(req: Request, res: Response): Promise<Response> {
+    try{
+        const userId = req.user?.id;
+  
+        if (!userId) {
+          return res.status(400).json({ message: 'User ID is required' });
+        }
+      
+        const cart = await Cart.findOne({
+          where: { buyerId: userId, status: 'active' },
+        });
+      
+        if (!cart) {
+          return res.status(404).json({ message: 'Cart not found' });
+        }
+    
+       await  Cart.update({status:"completed"}, {where:{ id:cart.id}})

```

Comment:
what if you use   await Cart.destroy({ where: { id: cart.id } }); so that it can also be removed from the database

Reply:
We can't because we will still need it when making an order